src/inputs/selectNext/selectNext.tsx

```diff
@@ -0,0 +1,439 @@
+import ClassNames from 'classnames';
+import {
+    isFunction,
+} from 'lodash';
+import PropTypes from 'prop-types';
+import React, {
+    LegacyRef,
+} from 'react';
+import ScrollBar from 'react-custom-scrollbars';
+import Select, {
+    components,
+    StylesConfig,
+} from 'react-select';
+import Icon from '../../dataDisplay/icon';
+import {
+    BEM_SELECT,
+    UI_CLASS_NAME,
+} from '../../global/constants';
+import makeStyles from '../../styles/makeStyles';
+
+type PropTypes = {
+    /**
+    * Assign a class name to the outer component.
+    */
+    className?: string;
+    /**
+    * Supply dropdown menu maximum height
+    */
+    dropdownMenuMaxHeight?: number;
+    /**
+    * Supply dropdown menu minimum height
+    */
+    dropdownMenuMinHeight?: number;
+    /**
+     * The `id` of the Select.
+     */
+    id?: string;
+    /**
+    * A Select can enable option search
+    */
+    isSearchable?: boolean;
+    /**
+     * The label for the Select.
+     */
+    label: string;
+    /**
+     * Whether the menu should use a portal, and where it should attach
+     * An example can be found in the Portaling documentation https://react-select.com/advanced#portaling
+     */
+    menuPortalTarget?: HTMLElement | null;
+    /**
+     * Text to display when there are no options
+     */
+    noOptionsMessage?: () => string | null;
+    /**
+     * Handle change events on the select
+     */
+    onChange: (nextValue: { value: string; label: string}) => void;
```

Comment:
We can use `Option` instead of `{ value: string; label: string}`, right?

Reply:
You're right!

Reply:
Looks good.  Resolving.